unit_tests/log_transfer/test_main_chkpt_info.cpp

```diff
@@ -0,0 +1,549 @@
+  env.after.unpack (unpac);
+  env.require_equal (env.before, env.after);
+
+  checkpoint_info after_2;
+
+  after_2.unpack (unpac);
+
+  env.require_equal (env.after, after_2);
+
+}
+
+test_env_chkpt::test_env_chkpt () : test_env_chkpt (100, 100)
+{
+}
+
+test_env_chkpt::test_env_chkpt (int size_trans, int size_sysops)
+{
+  srand (time (0));
+
+  log_lsa log_to_add = this->generate_log_lsa();
+  before.m_start_redo_lsa = log_to_add;
+
+  log_to_add = this->generate_log_lsa();
+  before.m_snapshot_lsa = log_to_add;
+
+  LOG_INFO_CHKPT_TRANS chkpt_trans_to_add;
+  LOG_INFO_CHKPT_SYSOP chkpt_sysop_to_add;
+
+  for (int i = 0; i < size_trans; i++)
+    {
+      chkpt_trans_to_add = generate_log_info_chkpt_trans();
+      before.m_trans.push_back (chkpt_trans_to_add);
+    }
+
+  for (int i = 0; i < size_sysops; i++)
+    {
+      chkpt_sysop_to_add = generate_log_info_chkpt_sysop();
+      before.m_sysops.push_back (chkpt_sysop_to_add);
+    }
+
+  before.m_has_2pc = rand() % 2;
+}
+
+test_env_chkpt::~test_env_chkpt ()
+{
+}
+
+LOG_INFO_CHKPT_TRANS
+test_env_chkpt::generate_log_info_chkpt_trans()
+{
+  LOG_INFO_CHKPT_TRANS chkpt_trans;
+
+  chkpt_trans.isloose_end = rand() % max;
+  chkpt_trans.trid        = rand() % max;
+  chkpt_trans.state       = static_cast<TRAN_STATE> (rand() % 15);
+
+  chkpt_trans.head_lsa    = generate_log_lsa();
+  chkpt_trans.tail_lsa    = generate_log_lsa();
+  chkpt_trans.undo_nxlsa  = generate_log_lsa();
+
+  chkpt_trans.posp_nxlsa  = generate_log_lsa();
+  chkpt_trans.savept_lsa  = generate_log_lsa();
+  chkpt_trans.tail_topresult_lsa  = generate_log_lsa();
+  chkpt_trans.start_postpone_lsa  = generate_log_lsa();
+
+  int length = rand() % LOG_USERNAME_MAX;
+
+  length = std::min (5, length);
+
+  for (int i = 0; i < length; i++)
+    {
+      chkpt_trans.user_name[i] = 'A' + rand() % 20;
+    }
+
+  return chkpt_trans;
+}
+
+LOG_INFO_CHKPT_SYSOP
+test_env_chkpt::generate_log_info_chkpt_sysop()
+{
+  LOG_INFO_CHKPT_SYSOP chkpt_sysop;
+
+  chkpt_sysop.trid                      = rand() % max;
+  chkpt_sysop.sysop_start_postpone_lsa  = generate_log_lsa();
+  chkpt_sysop.atomic_sysop_start_lsa    = generate_log_lsa();
+
+  return chkpt_sysop;
+}
+
+LOG_LSA
+test_env_chkpt::generate_log_lsa()
+{
+  LOG_LSA new_log = log_lsa (rand() % max, rand() % max);
+
+  while (std::find (used_logs.begin(), used_logs.end(), new_log) != used_logs.end())
+    {
+      new_log = log_lsa (rand() % max, rand() % max);
+    }
+
+  used_logs.push_back (new_log);
+  return new_log;
+}
+
+void
+test_env_chkpt::require_equal (checkpoint_info before, checkpoint_info after)
+{
+  REQUIRE (before.m_start_redo_lsa == after.m_start_redo_lsa);
+  REQUIRE (before.m_snapshot_lsa == after.m_snapshot_lsa);
+
+  REQUIRE (before.m_trans.size() == after.m_trans.size());
+  REQUIRE (before.m_trans == after.m_trans);
+
+  REQUIRE (before.m_sysops.size() == after.m_sysops.size());
+  REQUIRE (before.m_sysops == after.m_sysops);
```

Comment:
Vector equality operator includes size compare.
```suggestion
  REQUIRE (before.m_trans == after.m_trans);

  REQUIRE (before.m_sysops == after.m_sysops);
```

Reply:
They where added for debug purposes, removed now